packages/bruno-app/src/components/ResponsePane/QueryResult/QueryResultFilter/index.js

```diff
@@ -0,0 +1,45 @@
+import { IconFilter } from '@tabler/icons';
+import React, { useMemo } from 'react';
+import { Tooltip as ReactTooltip } from 'react-tooltip';
+
+const QueryResultFilter = ({ onChange, mode }) => {
+  const tooltipText = useMemo(() => {
+    if (mode.includes('json')) {
+      return 'Filter with JSONPath';
+    }
+
+    if (mode.includes('xml')) {
+      return 'Filter with XPath';
+    }
+
+    return null;
+  }, [mode]);
+
+  return (
+    <div className={'response-filter relative'}>
+      <div className="absolute inset-y-0 left-0 pl-4 flex items-center">
+        <div className="text-gray-500 sm:text-sm" id="request-filter-icon">
+          <IconFilter size={16} strokeWidth={1.5} />
+        </div>
+      </div>
+
+      {tooltipText && typeof tooltipText === 'string' && (
```

Comment:
You don't need the null & typeof check here. If `tooltipText` is null, the tooltip will not be shown.

Reply:
It's to hint the type to the IDE and be safe but I could also add   /** @type {string} */ to the useMemo line 6

![image](https://github.com/usebruno/bruno/assets/43440732/59b4c687-c202-480c-a2f8-35b8d31c67a3)